nexus/tests/integration_tests/instances.rs

```diff
@@ -1423,6 +1423,120 @@ async fn test_instance_failed_by_instance_watcher_automatically_reincarnates(
     );
 }
 
+// Verifies that if an instance transitions to `Failed` due to a failed request
+// to stop it, it will not automatically restart, even if it is configured to
+// restart it automatically on failure.
+#[nexus_test]
+async fn test_instance_failed_by_stop_request_does_not_reincarnate(
+    cptestctx: &ControlPlaneTestContext,
+) {
+    let apictx = &cptestctx.server.server_context();
+    let nexus = &apictx.nexus;
+    let client = &cptestctx.external_client;
+    let instance_name = "resurgam";
+    let instance_url = get_instance_url(instance_name);
+    let instance_id = dbg!(
+        make_forgotten_instance(
+            &cptestctx,
+            instance_name,
+            InstanceAutoRestartPolicy::BestEffort,
+        )
+        .await
+    );
+
+    // Attempting to stop the forgotten instance will result in a 404
+    // NO_SUCH_INSTANCE from the sled-agent, which Nexus turns into a 503.
+    expect_instance_stop_fail(
+        client,
+        instance_name,
+        http::StatusCode::SERVICE_UNAVAILABLE,
+    )
+    .await;
+
+    // Wait for the instance to transition to Failed.
+    dbg!(
+        instance_wait_for_state(client, instance_id, InstanceState::Failed)
+            .await
+    );
+
+    // Activate the reincarnation task.
+    dbg!(
+        nexus_test_utils::background::activate_background_task(
+            &cptestctx.internal_client,
+            "instance_reincarnation",
+        )
+        .await
+    );
+
+    // Unfortunately there isn't really a great way to assert "no start saga has
+    // been started", so we'll do the somewhat jankier thing of waiting a bit
+    // and making sure that the instance doesn't transition to Starting.
```

Comment:
How incredibly awful would it be to try to peer into the saga table omdb-style to try to look for one?